packages/opentelemetry/src/utils/getTraceData.ts

```diff
@@ -0,0 +1,40 @@
+import * as api from '@opentelemetry/api';
+import type { Span } from '@opentelemetry/api';
+import type { TraceData } from '@sentry/core';
+import { getCurrentScope } from '@sentry/core';
+import { getPropagationContextFromSpan } from '../propagator';
+import { generateSpanContextForPropagationContext } from './generateSpanContextForPropagationContext';
+
+/**
+ * Otel-specific implementation of `getTraceData`.
+ * @see `@sentry/core` version of `getTraceData` for more information
+ */
+export function getTraceData(span?: Span): TraceData {
```

Comment:
Would appreciate a review @mydea - I might be far off here. Basically what I need to get is a `Context` to pass to `propagation.inject`. I followed the code in our node fetch integration to get it but I'm not sure if this is correct: https://github.com/getsentry/sentry-javascript/blob/21830b1115bd85fcdc536f39a3153f874d725881/packages/node/src/integrations/node-fetch.ts#L84-L100

Reply:
Or IOW, maybe we can get rid of most code here and instead just somehow get to a `Context` directly from the `SpanContex of `spanToUse.spanContext()`?

Reply:
The tricky thing is getting the context from the span, if one is passed in. but it should be possible, you can get the scope via `getCapturedScopesOnSpan(span).scope`, and from there use `getContextFromScope(scope)` to get the context.

Then, we can really just do this:

```js
function getContextForSpan(span) {
  const scopes = getCapturedScopesOnSpan(span);
  if (scopes.scope) {
    return getContextFromScope(scopes.scope);
  }
  return undefined;
}

const ctx = (span && getContextForSpan(span)) || context.active();

const headers = {};
// this mutates the headers object
// this also handles it if there is no active span etc.
propagation.inject(ctx, headers, headerSetter);
return headers;
```